platform/security/idp/security-idp-server/src/main/java/org/codice/ddf/security/idp/server/IdpEndpoint.java

```diff
@@ -198,6 +256,115 @@ private void parseServiceProviderMetadata(List<String> serviceProviderMetadata)
 
     @POST
     @Path("/login")
+    @Consumes({"text/xml", "application/soap+xml"})
+    public Response doSoapLogin(InputStream body, @Context HttpServletRequest request) {
+        SoapBinding soapBinding = new SoapBinding(systemCrypto, serviceProviders);
+        try {
+            String bodyStr = IOUtils.toString(body);
+            AuthnRequest authnRequest = soapBinding.decoder()
+                    .decodeRequest(bodyStr);
+            String relayState = ((SoapRequestDecoder) soapBinding.decoder()).decodeRelayState(
+                    bodyStr);
+            soapBinding.validator()
+                    .validateRelayState(relayState);
+            soapBinding.validator()
+                    .validateAuthnRequest(authnRequest, bodyStr, null, null, null, strictSignature);
+            if (!request.isSecure()) {
+                throw new IllegalArgumentException("Authn Request must use TLS.");
+            }
+            boolean hasCookie = hasValidCookie(request, authnRequest.isForceAuthn());
+            AuthObj authObj = determineAuthMethod(bodyStr);
+            org.opensaml.saml.saml2.core.Response response = handleLogin(authnRequest,
+                    authObj.method, request, authObj, authnRequest.isPassive(), hasCookie);
+
+            Response samlpResponse = soapBinding.creator()
+                    .getSamlpResponse(relayState, authnRequest, response, null, soapMessage);
+            samlpResponse.getHeaders()
+                    .put("SOAPAction", Collections.singletonList(
+                            "http://www.oasis-open.org/committees/security"));
+            return samlpResponse;
+        } catch (IOException e) {
+            LOGGER.debug("Unable to decode SOAP AuthN Request", e);
+        } catch (SimpleSign.SignatureException e) {
+            LOGGER.debug("Unable to validate signature.", e);
+        } catch (ValidationException e) {
+            LOGGER.debug("Unable to validate request.", e);
+        } catch (SecurityServiceException e) {
+            LOGGER.debug("Unable to authenticate user.", e);
+        } catch (WSSecurityException | IllegalArgumentException e) {
+            LOGGER.debug("Bad request.", e);
+        }
+        //TODO return a soap fault here
+        return null;
+    }
+
+    private AuthObj determineAuthMethod(String bodyStr) {
+        XMLStreamReader xmlStreamReader = null;
+        try {
+            xmlStreamReader = xmlInputFactory.createXMLStreamReader(new StringReader(bodyStr));
+        } catch (XMLStreamException e) {
+            LOGGER.debug("Unable to parse SOAP message from client.", e);
+        }
+        SoapMessage soapMessage = new SoapMessage(Soap11.getInstance());
+        SAAJInInterceptor.SAAJPreInInterceptor preInInterceptor = new SAAJInInterceptor.SAAJPreInInterceptor();
+        soapMessage.setContent(XMLStreamReader.class, xmlStreamReader);
```

Comment:
Yeah I have to do this to do the work that the CXF stuff is doing internally. You don't normally use these things like I'm using them here.